src/helpers/fetchUrlsData.ts

```diff
@@ -1,22 +1,36 @@
-function fetchSingleUrlData(url: string): Promise<string> {
-  return fetch(url).then((r) => r.json());
+type RadarFetchError = {
+  description: string;
+  error: unknown;
+};
+
+type RadarResponse = string | RadarFetchError;
+
+async function fetchSingleUrlData(url: string): Promise<RadarResponse> {
+  try {
+    return fetch(url).then((r) => r.json());
+  } catch (e) {
```

Comment:
you can probably use the .catch() method.
mixing async, promise/A and try {} catch{} is confusing

Reply:
fair

Reply:
Switched it all to async due to personal preference. I find it more readable and I had to modify less code.